pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -30,32 +30,39 @@ import (
 	"k8s.io/apiserver/pkg/storage"
 	"k8s.io/apiserver/pkg/storage/names"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/dynamic-resource-allocation/structured"
-	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	"k8s.io/kubernetes/pkg/apis/resource"
 	"k8s.io/kubernetes/pkg/apis/resource/validation"
 	"k8s.io/kubernetes/pkg/features"
+	resourceutils "k8s.io/kubernetes/pkg/registry/resource"
 	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
 )
 
 // resourceclaimStrategy implements behavior for ResourceClaim objects
 type resourceclaimStrategy struct {
 	runtime.ObjectTyper
 	names.NameGenerator
+	nsClient v1.NamespaceInterface
 }
 
-// Strategy is the default logic that applies when creating and updating
-// ResourceClaim objects via the REST API.
-var Strategy = resourceclaimStrategy{legacyscheme.Scheme, names.SimpleNameGenerator}
+// NewStrategy is the default logic that applies when creating and updating ResourceClaim objects.
+func NewStrategy(ro runtime.ObjectTyper, ng names.NameGenerator, nsClient v1.NamespaceInterface) *resourceclaimStrategy {
```

Comment:
nit: I don't think we want to expose the flexibility of customizing ObjectTyper and NameGenerator outside this package, just limit the constructor parameter to `nsClient` and continue to hard-code legacyscheme.Scheme, names.SimpleNameGenerator internally